### PR TITLE
Add Chamber auto fan and Chamber servo operated vent

### PR DESCRIPTION
**Pull request by ladismrkolj**
### Requirements

Free ADC pin and one free PWM pins.

### Description

The idea behind is to have a "fully" air tight heated chamber, that has a servo operated vent (with(not yet implemented) or without dedicated fan), and a dedicated air heater with a fan.
The difference to other builds is that since the heated chamber is insulated and air tight the heated bed itself could overheat the chamber, so an automated vent is needed. However one can hardly predict when the heated bed itself will heat the heated chamber sufficiently and when additional air heater is needed.

I have added an option to change the chamber fan mode, currenty I only added two, but at least one is already drafted. First is completely static; the fan turns on as soon as the chamber target is above min_temp. second one is linear increase in fan speed, but only starts when the chamber temp is above chamber target (this one is best used when there is a fan that blows outside ari and there is no dedicated heater other than heated bed). Third is a V-shaped fan curve; the fan speed increases as the difference betwen actual chamber temperature and target temperature increases.

As for the servo operated vent. I wrote a simmilar algorithm to escaping from an infinite "wants_to_cool" loop. If the temperature of the heated chamber increases over HIGH_EXCESS_HEAT_LIMIT above the target temperature, then wait MIN_COOLING_SLOPE_TIME_CHAMBER_VENT seconds and if the temperature hasn't fallen at least MIN_COOLING_SLOPE_DEG_CHAMBER_VENT, then there is excess heat in the chamber, so the vent opens. The air heater is disabled and the vent automatically opens/closes depending on the temperature, however if something changes and the temperature falls LOW_EXCESS_HEAT_LIMIT bellow the target temperature, then the _flag_chamber_excess_heat_ is false and the heated chamber resumes normal operation with the vent closed.

 ### Benefits
Add the abbility to cool down the heated chamber and prevent overheating due to the heated bed. Also minimise the air movement in the heated chamber when the temperature is close to target temperature.

### Configurations

Note that my configuration.h has settings for serial that are set for my board.
The modifications are made in Configuration.h, Configuration_adv.h, temperature.cpp, Marlin/src/inc/SanityCheck.h and Marlin/src/pins/stm32f4/pins_BTT_SKR_PRO_common.h (however this should be coppied to all boards config.)

### Related Issues
Not yet implemented for all boards
/

### CODE REVIEW

File: Marlin/Configuration.h [hunk 1]
```diff
@@ -21,6 +21,8 @@
  */
 #pragma once
 
+#define CONFIG_EXAMPLES_DIR "Creality/Ender-5/Creality"
```
Comment (ladismrkolj) [inline Marlin/Configuration.h:24]: Only for Ender 5

File: Marlin/Configuration.h [hunk 2]
```diff
@@ -71,7 +73,7 @@
 // @section info
 
 // Author info of this build printed to the host during boot and M115
-#define STRING_CONFIG_H_AUTHOR "(none, default config)" // Who made the changes.
+#define STRING_CONFIG_H_AUTHOR "(ladismrkolj, Ender-5d)" // Who made the changes.
```
Comment (ladismrkolj) [inline Marlin/Configuration.h:76]: Custom changes

File: Marlin/Configuration.h [hunk 3]
```diff
@@ -574,6 +576,12 @@
 #define PREVENT_LENGTHY_EXTRUDE
 #define EXTRUDE_MAXLENGTH 200
 
+// @section CHAMBER
```
Comment (ladismrkolj) [inline Marlin/Configuration.h:579]: THIS IS RELEVANT FOR THE PULL REQUEST
adds the option to enable chamber fan and chamber vent

File: Marlin/Configuration_adv.h [hunk 4]
```diff
@@ -131,12 +133,35 @@
...
 #if TEMP_SENSOR_CHAMBER
   #define CHAMBER_MINTEMP             5
-  #define CHAMBER_MAXTEMP            60
+  #define CHAMBER_MAXTEMP            100
   #define TEMP_CHAMBER_HYSTERESIS     1   // (°C) Temperature proximity considered "close enough" to the target
   //#define CHAMBER_LIMIT_SWITCHING
   //#define HEATER_CHAMBER_PIN       44   // Chamber heater on/off pin
   //#define HEATER_CHAMBER_INVERTING false
-#endif
+  /**
```
Comment (ladismrkolj) [inline Marlin/Configuration_adv.h:141]: This are advanced settings for chamber fan and chamber vent

File: Marlin/Configuration.h (hunk 3 above)
Comment (thisiskeithb) [reply Marlin/Configuration.h:579]: You can't overwrite the default configs here with those for a specific printer. Just add changes relevant to the PR/feature and revert the rest.

You can submit a printer-specific config to the [Marlin Configurations repo](https://github.com/MarlinFirmware/Configurations/).

**Pull request closed without merging**